src/main/java/org/jenkins/plugins/lockableresources/actions/LockableResourcesRootAction.java

```diff
@@ -137,15 +148,43 @@ public int getFreeResourcePercentage(String label) {
    * @return All possible labels.
    */
   public Set<String> getAllLabels() {
-    return LockableResourcesManager.get().getAllLabels();
+    synchronized(lrm) {
+      return new HashSet<>(lrm.getAllLabels());
+    }
+  }
+
+  
+  public Set<AnLabel> getLabelsList() {
+    Set<AnLabel> list = new HashSet<>();
+    synchronized(lrm) {
+      for(String label : lrm.getAllLabels()) {
+        list.add(new AnLabel(label, lrm.getFreeResourceAmount(label), lrm.getResourcesWithLabel(label, null).size()));
+      }
+    }
+    return list;
+  }
+
+  public class AnLabel {
+    public AnLabel(String name, int free, int assigned) {
+      this.name = name;
+      this.free = free;
+      this.assigned = assigned;
+      this.percentage = (int)((double)free / (double)assigned * 100);
+    }
+    public String name;
+    public int free;
+    public int assigned;
+    public int percentage;
   }
 
   /**
    * Get amount of all labels.
    * @return Amount of all labels.
    */
   public int getNumberOfAllLabels() {
```

Comment:
check where it is used and set it to deprecated
This operation will be very slowly (because of synchronized over all resources)